tests/backends/pyq/test_time_dependent_generator.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+from typing import Any, Callable
+
+import numpy as np
+import pytest
+import qutip
+import torch
+from metrics import MIDDLE_ACCEPTANCE
+
+from qadence import (
+    AbstractBlock,
+    HamEvo,
+    Parameter,
+    QuantumCircuit,
+    QuantumModel,
+    Register,
+    TimeParameter,
+    X,
+    Y,
+)
+
+
+@pytest.fixture
+def omega() -> float:
+    return 20.0
+
+
+@pytest.fixture
+def feature_param_value() -> float:
+    return 2.0
+
+
+@pytest.fixture
+def qadence_generator(omega: float) -> AbstractBlock:
+    t = TimeParameter("t")
+    x = Parameter("x", trainable=False)
+    generator_t = omega * (t * X(0) + x * (t**2) * Y(1))
+    return generator_t  # type: ignore [no-any-return]
+
+
+@pytest.fixture
+def qutip_generator(omega: float, feature_param_value: float) -> Callable:
+    def generator_t(t: float, args: Any) -> qutip.Qobj:
+        return omega * (
+            t * qutip.tensor(qutip.sigmax(), qutip.qeye(2))
+            + feature_param_value * t**2 * qutip.tensor(qutip.qeye(2), qutip.sigmay())
+        )
+
+    return generator_t
+
+
+def test_time_dependent_generator(
```

Comment:
It would be good to expand tests to a randomly generated generator if possible.

Reply:
Regarding this - it might be not that easy to do that since I'll have to also create some kind of parser that creates a Hamiltonian conforming to `qutip`'s requirements. It's possible but it's a bit time consuming.

Reply:
OK got it.